File: solutions/python/allergies/1/allergies.py
import numpy as np

allergies = np.array([
    "eggs",
    "peanuts",
    "shellfish",
    "strawberries",
    "tomatoes",
    "chocolate",
    "pollen",
    "cats"
])

class Allergies:

    def __init__(self, score):
        bin_arr = np.flip(np.array(list(np.binary_repr(score % 256)), dtype=int))
        self._allergies = allergies[np.where(bin_arr == 1)[0]]

    def allergic_to(self, item):
        return item in self._allergies

    @property
    def lst(self):
        return self._allergies.tolist()

File: solutions/javascript/allergies/1/allergies.js
//
// This is only a SKELETON file for the 'Allergies' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const allergies = ["eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"];

export class Allergies {
  constructor(result) {
    this.result = result;
  }

  list() {
    return allergies.filter(allergy => this.allergicTo(allergy));
  }

  allergicTo(allergy) {
    return (this.result & (1 << allergies.indexOf(allergy))) !== 0;
  }
}
